MonoGame.Framework/Input/GamePadState.cs

```diff
@@ -87,6 +87,20 @@ public GamePadState(Vector2 leftThumbStick, Vector2 rightThumbStick, float leftT
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:Microsoft.Xna.Framework.Input.GamePadState"/> struct.
+        /// using the specified stick, trigger, and button values.
+        /// </summary>
+        /// <param name="leftThumbStick">Left stick value. Each axis is clamped between −1.0 and 1.0.</param>
+        /// <param name="rightThumbStick">Right stick value. Each axis is clamped between −1.0 and 1.0.</param>
+        /// <param name="leftTrigger">Left trigger value. This value is clamped between 0.0 and 1.0.</param>
+        /// <param name="rightTrigger">Right trigger value. This value is clamped between 0.0 and 1.0.</param>
+        /// <param name="buttons"> Array or parameter list of Buttons to initialize as pressed.</param>
+        public GamePadState(Vector2 leftThumbStick, Vector2 rightThumbStick, float leftTrigger, float rightTrigger, params Buttons[] buttons)
+            : this(new GamePadThumbSticks(leftThumbStick, rightThumbStick), new GamePadTriggers(leftTrigger, rightTrigger), new GamePadButtons(buttons), new GamePadDPad(buttons[0]))

```

Comment:
The last parameter `new GamePadDPad(buttons[0])` is not correct. All buttons in the array should be considered, not just the first one.

The right way would be to recreate the constructor in `GamePadDPad` that takes an array. This constructor previously existed and can be seen here: https://github.com/MonoGame/MonoGame/pull/4970/files#diff-aea906acf1cef920f135ebf0a0ad686dL80
When you recreate it, it does not to take `params Buttons[]`, but just `Buttons`.